framework/src/util/CFRLogger.cpp

```diff
@@ -30,6 +29,16 @@ namespace cppmicroservices
     {
         CFRLogger::CFRLogger() : serviceTracker(), logService(nullptr) {}
 
+	CFRLogger::CFRLogger(cppmicroservices::BundleContext context)
+            : cfrContext(context)
+            , serviceTracker(
```

Comment:
If `CFRLogger::Open()` is called after this constructor runs, then `serviceTracker` is destroyed and re-created as a new service tracker. This is a redundant operation.

Reply:
I added this constructor since without this, the tests were failing (With error: Actual function call count doesn't match EXPECT_CALL's). I am guessing that since we were not opening service tracker in the constructor, the calls to Log in the LogService were not being made.

I have also tried calling default constructor of CFRLogger and then Open() method on the object, but I am getting same error in the test.

Please let me know if there can be some other way to avoid doing this and run the tests. Thanks.

Reply:
@architapruthi, which test is failing?

Reply:
The tests are running fine with the new constructor.
I will see if I can achieve running the tests without adding that. Thanks.

Reply:
When I use the default constructor and Open() call, the cfrContext object (bundle context) is null only and hence the service tracker is not starting to start tracking the service and call the Log methods.

Hence, I have kept the new constructor for now, for testing purposes.

Reply:
Made the changes as discussed.